File: src/model/payload.ts
import {StockHistData, StockHistMinData} from "./stock.ts";

export type IPayload =

    | {
    event: 'dashboard';
    data: StockHistData;
    msgId? : string;
}
    | {
    event: 'kLineRT';
    data: StockHistMinData[];
    msgId? : string;
    timestamp?: number;
    isFull?: number;
}

    | {
    event: 'started';
    data: {
        url: string;
        downloadId: number;
        contentLength: number;
    };
}
    | {
    event: 'progress';
    data: {
        downloadId: number;
        chunkLength: number;
    };
}
    | {
    event: 'finished';
    data: {
        downloadId: number;
    };
};

File: src-tauri/src/model/payload.rs
use crate::model::{StockHist, StockTimeSharing};

///
/// c2s
///
#[derive(serde::Serialize, serde::Deserialize, Clone)]
pub struct InitializePayload{
    #[serde(rename = "msgId")]
    pub msg_id: Option<i64>,
}

#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
#[serde(rename_all = "camelCase", tag = "event")]
pub enum IPayload {
    Dashboard{data: StockHist, msg_id:Option<String>},
    KLine{ data: StockHist,  data_type:String, msg_id:Option<String>, timestamp:Option<i64>},
    KLineRT{data: Vec<StockTimeSharing>,
        #[serde(rename = "isFull")]
        is_full:u8,
        #[serde(rename = "msgId")]
        msg_id:Option<String>,
        timestamp:Option<i64>
    },
    CloseChannel,
}

